tests/integration/test_db_migration.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Discourse integration tests."""
+
+import logging
+
+import pytest
+from botocore.config import Config
+from ops.model import WaitingStatus
+from pytest_operator.plugin import Model, OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_db_migration(model: Model, ops_test: OpsTest, pytestconfig: Config, run_action):
+    """
+    arrange: preload postgres with a mock db that is created in Discource v3.2.0
+    act: deploy and integrate with discourse v3.3.0
+    assert: discourse must be active idle, previously it was creating migration
+    errors related to not being able to delete some columns because of triggers
+    """
+    postgres_app = await model.deploy(
+        "postgresql-k8s",
+        channel="14/stable",
+        series="jammy",
+        revision=300,
+        trust=True,
+        config={"profile": "testing"},
+    )
+    async with ops_test.fast_forward():
+        await model.wait_for_idle(apps=[postgres_app.name], status="active")
+    await postgres_app.set_config(
+        {
+            "plugin_hstore_enable": "true",
+            "plugin_pg_trgm_enable": "true",
+        }
+    )
+    await model.wait_for_idle(apps=[postgres_app.name], status="active")
+    db_pass = await run_action(postgres_app.name, "get-password", username="operator")
+    db_pass = db_pass["password"]
+    return_code, _, scp_err = await ops_test.juju(
+        "scp",
+        "--container",
+        "postgresql",
+        "./testing_database/testing_database.sql",
+        f"{postgres_app.units[0].name}:.",
+    )
+
+    assert return_code == 0, scp_err
+
+    return_code, _, ssh_err = await ops_test.juju(
+        "ssh",
+        "--container",
+        "postgresql",
+        postgres_app.units[0].name,
+        "createdb -h localhost -U operator --password discourse",
+        stdin=str.encode(f"{db_pass}\n"),
+    )
+    assert return_code == 0, ssh_err
+
+    return_code, _, ssh_err = await ops_test.juju(
+        "ssh",
+        "--container",
+        "postgresql",
+        postgres_app.units[0].name,
+        "pg_restore -h localhost -U operator\
+              --password -d discourse\
+                  --no-owner --clean --if-exists ./testing_database.sql",
+        stdin=str.encode(f"{db_pass}\n"),
+    )
```

Comment:
I still feel like we're missing something that gives us confidence the database is in the state we want it to be. Is there a "patch" table or "migration" table or something that records what version of the DB we are currently running, and could we query that to confirm that we have the same version as expected for 3.2.0?

Reply:
Searched through the db but was not able to find anything about the version. Maybe we can deploy v3.2.0 with this db to show it doesn't create any errors. Wdyt?

Reply:
What about the `schema_migrations` and `schema_migration_details` tables?

Reply:
They specify the migrations that happened that are listed in here: https://github.com/discourse/discourse/tree/main/db/migrate

They don't specify the discourse version but I assume we can get the latest migration file from there and maybe confirm its the v3.2.0